cmd/cvp-trigger/main.go

```diff
@@ -1,72 +1,183 @@
 package main
 
 import (
+	"encoding/json"
 	"flag"
 	"fmt"
 	"os"
+	"strings"
 
 	"github.com/ghodss/yaml"
+	"github.com/openshift/ci-tools/pkg/util"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/client-go/rest"
-
 	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
 	pjclientset "k8s.io/test-infra/prow/client/clientset/versioned"
 	prowconfig "k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/interrupts"
 	"k8s.io/test-infra/prow/pjutil"
-
-	"github.com/openshift/ci-tools/pkg/util"
+	"k8s.io/test-infra/prow/pod-utils/decorate"
 )
 
 const (
-	prowConfigPathOption = "prow-config-path"
-	jobConfigPathOption  = "job-config-path"
-	periodicOption       = "periodic"
+	bundleImageRefOption       = "bundle-image-ref"
+	channelOption              = "channel"
+	indexImageRefOption        = "index-image-ref"
+	installNamespaceOption     = "install-namespace"
+	jobConfigPathOption        = "job-config-path"
+	jobNameOption              = "job-name"
+	ocpVersionOption           = "ocp-version"
+	operatorPackageNameOptions = "operator-package-name"
+	outputFilePathOption       = "output-path"
+	prowConfigPathOption       = "prow-config-path"
+	releaseImageRefOption      = "release-image-ref"
+	targetNamespacesOption     = "target-namespaces"
 )
 
 type options struct {
-	confirm bool
+	bundleImageRef      string
+	channel             string
+	indexImageRef       string
+	installNamespace    string
+	jobConfigPath       string
+	jobName             string
+	ocpVersion          string
+	operatorPackageName string
+	outputPath          string
+	prowConfigPath      string
+	releaseImageRef     string
+	targetNamespaces    string
+	dryRun              bool
+}
 
-	prowConfigPath string
-	jobConfigPath  string
+type jobResult interface {
+	toJSON() ([]byte, error)
+}
 
-	periodic string
+type prowjobResult struct {
+	Status       pjapi.ProwJobState `json:"status"`
+	ArtifactsURL string             `json:"prowjob_artifacts_url"`
+	URL          string             `json:"prowjob_url"`
 }
 
-func gatherOptions() options {
-	o := options{}
-	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+func (p *prowjobResult) toJSON() ([]byte, error) {
+	j, err := json.MarshalIndent(p, "", "    ")
```

Comment:
This method should just be `return json.MarshalIndent(...` and at that point I don't think there's any utility in a `toJSON` method at all

Reply:
Updated to `return json.MarshalIndent(...`
Method exists to allow for unit testing with a fake result.